src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
> However, I didn't think it would be good to rely on the structure of this string.

@weshaggard, without exposing version information, how are developers supposed to do this kind of thing?  I keep hearing "they should instead use feature detection"... how does that work in a case like this?